src/main/java/org/cactoos/list/Immutable.java

```diff
@@ -55,10 +66,49 @@
 
     /**
      * Ctor.
-     * @param src Source
+     * @param items Source array
+     */
+    @SafeVarargs
+    public Immutable(final T... items) {
+        this(new IterableOf<>(items));
+    }
+
+    /**
+     * Ctor.
+     * @param src Source collection
+     */
+    public Immutable(final Collection<T> src) {
+        this(new IterableOf<>(src.iterator()));
+    }
+
+    /**
+     * Ctor.
+     * @param src Source list
      */
     public Immutable(final List<T> src) {
-        this.list = src;
+        this(new IterableOf<>(src.iterator()));
+    }
+
+    /**
+     * Ctor.
+     * @param src Source iterable
+     */
+    public Immutable(final Iterable<T> src) {
+        this(() -> {
```

Comment:
@iakunin please, use `CollectionOf` instead of `IterableOf`.

Reply:
@fabriciofx fixed